### ISSUE TITLE
Evaluating/transpiling files with no default export

### ISSUE THREAD

**Issue by ORESoftware** [Microsoft/TypeScript#13381]
**TypeScript Version:**

$ tsc -v
Version 2.1.4

**Code**

```js

// this .js file is just an executable, it has no exports

#!/usr/bin/env node

console.log('foo bar baz')

```

**Expected behavior:**

I would export that TS would handle this file and just transpile it without worrying about imports/exports.
In this case, it's a .js file, so we would just be copying it from the source location to the outDir location. But if it were a .ts file with no default exports, I would hope TS would transpile it without complaining.

**Actual behavior:**

I get this error:

`test/one.test.js(77,9): error TS7027: Unreachable code detected.`
It says that the file has no default export so it can't transpile it .

What I am looking to do is transpile (in this case just copy) .js files from the source dir to the outDir, but TS is complaining when the files have no default export. Is there a setting with .tsconfig that I can use so that TS will not complain/error when I point TS at either a .ts file or .js that does not have a default export?

**Comment by ORESoftware** [comment]
I looked through these options, many of which I assume are applicable to .tsconfig, and I don't think any of them apply to this use case?

https://www.typescriptlang.org/docs/handbook/compiler-options.html

**Comment by ORESoftware** [comment]
oh wait! perhaps this is the one:

`--allowUnreachableCode`

**Comment by ORESoftware** [comment]
that seemed to work, will close
